source/common/config/grpc_stream.h

```diff
@@ -153,6 +225,12 @@ class GrpcStream : public Grpc::AsyncStreamCallbacks<ResponseProto>,
   TokenBucketPtr limit_request_;
   const bool rate_limiting_enabled_;
   Event::TimerPtr drain_request_timer_;
+
+  // Records the initial message and timestamp of the most recent remote closes with the same
```

Comment:
I recommend prepending `last_` or if you think it's more clear `last_recorded_` to each of the following variables.